packages/snowpack/src/config.ts

```diff
@@ -551,6 +527,43 @@ function normalizeProxies(proxies: RawProxies): Proxy[] {
   });
 }
 
+function normalizeMount(config: SnowpackConfig) {
+  const mountedDirs: Record<string, string> = config.mount || {};
+  for (const [target, cmd] of Object.entries(config.scripts)) {
+    if (target.startsWith('mount:')) {
+      const cmdArr = cmd.split(/\s+/);
+      if (cmdArr[0] !== 'mount') {
+        handleConfigError(`scripts[${target}] must use the mount command`);
+      }
+      cmdArr.shift();
+      const {to, _} = yargs(cmdArr);
+      if (_.length !== 1) {
+        handleConfigError(`scripts[${target}] must use the format: "mount dir [--to /PATH]"`);
+      }
+      if (target === 'mount:web_modules') {
+        config.buildOptions.webModulesUrl = to;
+      } else {
+        mountedDirs[cmdArr[0]] = to || `/${cmdArr[0]}`;
+      }
+    }
+  }
+  for (const [mountDir, mountUrl] of Object.entries(mountedDirs)) {
+    const fromDisk = path.posix.normalize(mountDir + '/');
+    delete mountedDirs[mountDir];
+    mountedDirs[fromDisk] = mountUrl;
+    if (mountUrl[0] !== '/') {
+      handleConfigError(
+        `mount[${mountDir}]: Value "${mountUrl}" must be a URL path, and start with a "/"`,
+      );
+    }
+  }
+  // if no mounted directories, mount the root directory to the base URL
+  if (!Object.keys(mountedDirs).length) {
+    mountedDirs['.'] = '/';
+  }
+  return mountedDirs;
+}
```

Comment:
I have been wondering if the default behavior of the project root being default mount is helpful. In most JS apps, a `src` directory is pretty standard. But people do use different directories for static files, like `public` or `static` so that’s where the opinion splinters. Still, I almost feel like the way the templates are set up now should be the default:

```json
"mount": {
  "src": "/_dist_",
  "public": "/"
}
```

And people could opt-out and mount the root with `{ "mount": { ".": "/" } }` if they wanted (which I think is less-standard?).

Reply:
Yea, I was thinking about that as well. Kind of like how 11ty forces you to explicitly call out a directory as "pass-through". But it definitely ruins the zero config story a bit. Might be worth revisiting for v3.